src/DonateHope.Core/Services/CampaignContributionServices/CampaignContributionDeletingService.cs

```diff
@@ -0,0 +1,61 @@
+using DonateHope.Core.DTOs.CampaignContributionDTOs;
+using DonateHope.Core.Errors;
+using DonateHope.Core.Mappers;
+using DonateHope.Core.ServiceContracts.CampaignContributionsServiceContracts;
+using DonateHope.Domain.RepositoryContracts;
+using FluentResults;
+
+namespace DonateHope.Core.Services.CampaignContributionServices;
+
+public class CampaignContributionDeletingService(
+    ICampaignContributionsRepository campaignContributionsRepository,
+    CampaignContributionMapper campaignContributionMapper
+    ) : ICampaignContributionDeletingService
+{
+    private readonly ICampaignContributionsRepository _campaignContributionsRepository = campaignContributionsRepository;
+    private readonly CampaignContributionMapper _campaignContributionMapper = campaignContributionMapper;
+
+    public async Task<Result<CampaignContributionDeleteResponseDto>> DeleteCampaignContributionAsync(
+        Guid campaignContributionId,
+        Guid deletedBy,
+        String reasonForDeletion 
+        )
+    {
+        // Check if reason for deletion is provided
+        if (string.IsNullOrWhiteSpace(reasonForDeletion))
```

Comment:
Just curious, why we use IsNullOrWhiteSpace over IsNullOrEmnpty here?

Reply:
Yeah, let check these cases:
1. String.IsNullOrEmpty(""); 
2. String.IsNullOrEmpty(null); 
3. String.IsNullOrEmpty("   ");
4. String.IsNullOrEmpty("
");

- `IsNullOrEmnpty`  will return `true` for 2 cases: 1, 2
- `IsNullOrWhiteSpace` will return `true` for all 4 cases

Reply:
Moved this check to validator